### ISSUE TITLE
Flatten helper

### ISSUE THREAD

**Issue by poteto** [DockYard/ember-composable-helpers#32]
Deep flatten. 

```js
flatten([1], [2], [3]) === [1, 2, 3];
flatten([1], [2, 3, [4, 5]]) === [1, 2, 3, 4, 5];
```

```hbs
{{#each (flatten (repeat 3 (range 1 3 true))) as |number|}}
  {{! renders something like `1 2 3 1 2 3 1 2 3`}}
{{/each}}
```

```hbs
{{#each (flatten (repeat 3 (take 3 items))) as |number|}}
  {{! renders something like `first second third first second third first second third`}}
{{/each}}
```

**Comment by zigahertz** [comment]
it looks @vikram7 had a PR for this, but it got closed; is there any plan to keep working on that branch? I could use a `flatten` helper myself...

**Comment by martndemus** [comment]
@zigahertz feel free to open a fresh PR!

**Comment by Asherlc** [comment]
See https://github.com/DockYard/ember-composable-helpers/pull/158

**Comment by martndemus** [comment]
Implemented by #158
